pymeasure/instruments/agilent/agilentE4407B.py

```diff
@@ -0,0 +1,637 @@
+        auto mode. This property can be set.
+        """,
+    )
+    max_mixer_power = Instrument.control(
+        ":SENS:POW:MIX:RANG?",
+        ":SENS:POW:MIX:RANG %g",
+        """ A floating point property that represents the maximum mixer power
+        in dBm. This property can be set.
+        """,
+    )
+    # system commands
+
+    def reset(self):
+        """
+        A command that resets the instrument.
+        """
+        self.write("*RST")
+
+    def preset(self):
+        """Reset the instrument to the preset conditions."""
+        self.write("SYST:PRES")
+
+    def persistant_reset(self):
+        """Reset the instrument persistant state vakues to there factory defaults."""
+        self.write("SYST:PRES:PERS")
+
+    def save_user_preset(self):
+        """Save the current instrument state as a user preset."""
+        self.write("SYST:PRES:USER:SAVE")
+
+    def abort(self):
+        """
+        A command that aborts the sweep or measurement in progress.
+        """
+        self.write("ABOR")
+
+    def hardware_configuration(self):
+        """
+        A command that returns information about the current hardware in the instrument.
+        """
+        self.write("SYS:CONF:HARD?")
+
+    def system_configuration(self):
+        """
+        A command that returns information about the configuration in the instrument.
+        """
+        self.write("SYS:CONF:SYST?")
+
+    recall = Instrument.setting(
+        "*RCL %g",
+        """Recall the instrument state from specified memory resgister.""",
+        validator=strict_discrete_range,
+        values=[2, 30],
+    )
+    SAVE = Instrument.setting(
+        "*SAV %g",
+        """Save the instrument state to specified memory register.""",
+        validator=strict_discrete_range,
+        values=[2, 30],
+    )
+
+    time = Instrument.control(
+        ":SENS:SWE:TIME?",
+        ":SENS:SWE:TIME %g",
+        """Set the real time clock of the instrument. <hours>,<minutes>,<seconds>""",
+    )
+
+    date = Instrument.control(
+        "SYS:DATE?",
+        "SYS:DATE %g",
+        """Set the date of the instrument. <year>,<month>,<day> ####,##,##""",
+    )
+
+    error_queue = Instrument.measurement(
+        ":SYST:ERR?",
+        """A string property that returns the error queue of the instrument.
+        """,
+    )
+
+    options = Instrument.measurement(
+        "SYS:OPT?",
+        """A string property that returns the options of the instrument.
+        """,
+    )
+
+    preset_type = Instrument.setting(
+        ":SYST:PRES:TYPE %g",
+        """ A string property that represents the preset type.
+         can be set to FACT, USER, or MODE.
+         """,
+        validator=strict_discrete_set,
+        values=["FACT", "USER", "MODE"],
+    )
+    # Calibration commands
+    align_all = Instrument.control(
+        ":CAL:ALL?",
+        ":CAL:ALL",
+        """align all the circuits of the instrument.
```

Comment:
this feels like an action. what happens when you do the get part (":CAL:ALL?") - what does that return?